Expensify/App
# Remove null values when calling Ion.merge() batch merge updates

cc @quinthar @tgolen 

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/145190

### Tests
1. Verify data is able to be set and merged normally
1. Run this test to verify that `Ion.merge('someKey', {test: null})` can remove a key safely

e.g. add this diff to `Expensify.js`

```diff
diff --git a/src/Expensify.js b/src/Expensify.js
index 56623e01..59c516f1 100644
--- a/src/Expensify.js
+++ b/src/Expensify.js
@@ -10,6 +10,20 @@ import IONKEYS from './IONKEYS';
 import withIon from './components/withIon';
 import styles from './styles/StyleSheet';
 import Log from './libs/Log';
+import AsyncStorage from '@react-native-community/async-storage';
+
+Ion.merge('testing', {1: '1'});
+Ion.merge('testing', {2: '2'});
+Ion.merge('testing', {3: '3'});
+Ion.merge('testing', {4: '4'});
+
+Ion.merge('testing', {1: null});
+Ion.merge('testing', {2: null});
+
+setTimeout(() => {
+    AsyncStorage.getItem('testing')
+        .then(result => console.log(result));
+}, 2000);
```
and verify you see that the keys have been removed and an object like this remains `{3: '3', 4: '4'}`

### Screenshots
❌
